### ISSUE TITLE
Allow call sheets to be copied and modified

### ISSUE THREAD

**Issue by timriley** [dry-rb/dry-transaction#6]
@apotonick has raised a use case in which he'd set up a "base" call sheet of some kind and then modify it based on certain runtime conditions.

Adding support for call sheets to be copied and modified seems like it could be done fairly nicely, without affecting any of the existing API.

Something like this for adding new steps:

``` ruby
my_trans = CallSheet(container: container) do
  step :my_step, "my_app.my_step"
end

another_trans = my_trans.insert(before :my_step) do
  step :another_step, "my_app.another_step"
end
```

And for removing steps:

```
another_trans = my_trans.remove(:my_step)
```

Both of these methods would make a copy of the existing call sheet before making any adjustments.

**Comment by apotonick** [comment]
:heart:

**Comment by apotonick** [comment]
Assuming that the block is simply invoked in the callsheet context, you may add multiple steps in one `insert` go, right?

**Comment by timriley** [comment]
Yeah, that'd be the plan.

**Comment by timriley** [comment]
Resolved by #7
